app/components/test-panel/test-panel.component.ts

```
@@ -4,16 +4,105 @@
 
 
 import { Component, Input, Output } from "@angular/core";
-
+import { Program } from "../../models/plugin";
 
 @Component({
     selector: "sinap-test-panel",
     templateUrl: "./test-panel.component.html",
-    styleUrls: ["../../styles/bottom-panel.component.css"]
+    styleUrls: ["./test-panel.component.css", "../../styles/bottom-panel.component.css"]
 })
 export class TestPanelComponent {
-    @Output() @Input() tests = [['001001', true, true],
-    ['001010', false, false],
-    ['101010101', true, false],
-    ['100100011', true, true]];
+    private _program: Program | null = null;
+
+    set program(program: Program | null) {
+        this._program = program;
+
+        if (program && this.autoplay) {
+            this.runTests();
+        }
+    }
+
+    get program() {
+        return this._program;
+    }
+
+    private tests: Test[] = [];
+    private selected = new Set<Test>();
+
+    /**
+     * Whether changes to the graph should automatically run through all tests.
+     */
+    private autoplay: Boolean = true;
+
+
+
+    private runTests() {
+        this.tests.forEach((test) => {
+            this.runTest(test);
+        });
+    }
+
+    private runTest(test: Test) {
+        if (this.program) {
+            this.program.run(test.input).then((output) => {
+                console.log(test, output);
+                test.output = output as boolean;
+            }).catch((error) => {
+                test.output = null;
```

You should set test.output = error instead of swallowing the error.